pkg/reconciler/configuration/resources/revision.go

```diff
@@ -65,6 +62,19 @@ func UpdateRevisionLabels(rev *v1alpha1.Revision, config *v1alpha1.Configuration
 	}
 }
 
+// UpdateRevisionAnnotations sets the revisions annotations given a Configuration's updater annotation.
+func UpdateRevisionAnnotations(rev *v1alpha1.Revision, config *v1alpha1.Configuration) {
+	if rev.Annotations == nil {
+		rev.Annotations = make(map[string]string)
+	}
+
+	// Populate the CreatorAnnotation from configuration.
+	cans := config.GetAnnotations()
+	if c, ok := cans[serving.UpdaterAnnotation]; ok {
+		rev.GetAnnotations()[serving.CreatorAnnotation] = c
```

Comment:
nit: I'd use `rev.Annotations[serving.CreatorAnnotation] = c` here. You're directly accessing the field above anyway and `GetAnnotations()` reads like it might return a copy.

That might only be me though, feel free to ignore this.